src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.AssemblyResolution.targets

```diff
@@ -67,8 +71,8 @@ _ResolveAssemblies MSBuild target.
         Condition=" '%(Identity)' != '' "
     />
     <ProcessAssemblies
-        InputAssemblies="@(_ResolvedAssemblyFiles)"
-        ResolvedSymbols="@(_ResolvedSymbolFiles)"
+        InputAssemblies="@(_ResolvedAssemblyFiles->Distinct())"
```

Comment:
Is `@(_ResolvedAssemblyFiles->Distinct())` potentially stored in a file anywhere? Presumably this expression controls `@(_ProcessedShrunkAssemblies)`? `.Distinct()` may change the ordering, and I don't know if that ordering is consistent from one process run to the next, so if this is "somehow" stored in a file anywhere, it may result in bad dependency tracking.

Reply:
`->Distinct()` in MSBuild preserves ordering, I asked Rainer about this before:

https://github.com/dotnet/msbuild/blob/6605cdf6e969fa670bab736a0aca415da2080cdb/src/Build/Evaluation/Expander.cs#L2517-L2555

`@(_ResolvedAssemblyFiles)` won't be used anywhere else, but the outputs of `<ProcessAssemblies/>` will be.